pkg/standalone-utils/contracts/HyperEVMRateProvider.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {
+    IHyperEVMRateProvider
+} from "@balancer-labs/v3-interfaces/contracts/standalone-utils/IHyperEVMRateProvider.sol";
+import { IRateProvider } from "@balancer-labs/v3-interfaces/contracts/solidity-utils/helpers/IRateProvider.sol";
+
+import { HyperSpotPricePrecompile } from "./utils/HyperSpotPricePrecompile.sol";
+import { HyperTokenInfoPrecompile } from "./utils/HyperTokenInfoPrecompile.sol";
+
+contract HyperEVMRateProvider is IRateProvider, IHyperEVMRateProvider {
+    uint256 private immutable _spotPriceDivisor;
+    uint32 private immutable _pairIndex;
+    uint32 private immutable _tokenIndex;
+
+    constructor(uint32 tokenIndex, uint32 pairIndex) {
+        uint8 szDecimals = HyperTokenInfoPrecompile.szDecimals(tokenIndex);
+        _spotPriceDivisor = 10 ** (8 - szDecimals);
```

Comment:
Presumably we don't need to validate this because the Hyperliquid protocol guarantees documents have a max of 8 decimals? Could document this.